ledger/src/blockstore_meta.rs

```diff
@@ -526,6 +526,23 @@ impl OptimisticSlotMetaVersioned {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SizeDetails {
+    pub last_fec_set_index: u32,
+    /// These fields keep track of any shreds inserted into blockstore,
+    /// through turbine/repair or erasure recovery.
+    /// Before voting replay will check that the sum of these values are at
+    /// least 2 * `DATA_SHREDS_PER_FEC_BLOCK`.
```

Comment:
I thought we wanted to make sure that the leader broadcasted enough many shreds of the last FEC set.

With this accounting, the leader can generate a 1:63 erasure batch (one data shred and 63 coding shred), send only 1 coding shred to a node, and that node will be able to recover the whole batch and insert the remaining shreds.

Reply:
ah yeah, that's an excellent point. we'll need to differentiate the source of the shreds. 
let me start a discussion to figure out the requirement.